ddf-rpc-api/src/main/java/com/connexta/ddf/platform/direct/PlatformMethods.java

```diff
@@ -0,0 +1,174 @@
+package com.connexta.ddf.platform.direct;
+
+import static com.connexta.util.MapFactory.mapOf;
+import static java.util.Arrays.asList;
+
+import com.connexta.ddf.catalog.direct.CatalogMethods;
+import com.connexta.jsonrpc.Error;
+import com.connexta.jsonrpc.MethodSet;
+import com.connexta.jsonrpc.RpcFactory;
+import com.connexta.jsonrpc.RpcMethod;
+import com.connexta.jsonrpc.impl.RpcFactoryImpl;
+import com.connexta.util.ImmutablePair;
+import ddf.action.Action;
+import ddf.action.ActionRegistry;
+import ddf.action.impl.ActionImpl;
+import ddf.catalog.CatalogFramework;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.data.types.Core;
+import ddf.catalog.federation.FederationException;
+import ddf.catalog.filter.FilterBuilder;
+import ddf.catalog.operation.QueryRequest;
+import ddf.catalog.operation.QueryResponse;
+import ddf.catalog.operation.impl.QueryImpl;
+import ddf.catalog.operation.impl.QueryRequestImpl;
+import ddf.catalog.source.SourceUnavailableException;
+import ddf.catalog.source.UnsupportedQueryException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.opengis.filter.sort.SortBy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PlatformMethods implements MethodSet {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(PlatformMethods.class);
+  private static final int ACTION_ID_QUERY_LIMIT = 500;
+
+  private final RpcFactory rpc = new RpcFactoryImpl();
```

Comment:
✏️ Consider injecting this

Reply:
This I do plan on doing but will probably save for another PR